pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -197,8 +197,10 @@ func (rc *reconciler) reconcile() {
 			}
 			// Check whether timeout has reached the maximum waiting time
 			timeout := elapsedTime > rc.maxWaitForUnmountDuration
-			// Check whether volume is still mounted. Skip detach if it is still mounted unless timeout
-			if attachedVolume.MountedByNode && !timeout {
+			safeToDetach := rc.desiredStateOfWorld.SafeToDetach(attachedVolume.NodeName)
+
```

Comment:
When the node is added back (or a new node with the same name joins the cluster), SafeToDetach will still show it is safe to detach. But it is might not true any more.